### ISSUE TITLE
Nested array with incomplete objects leads to type being lost

### ISSUE THREAD

**Issue by coli** [nijikokun/generate-schema#22]
If an array contains a empty or partial object, the type inferred is lost. Example:
```
require("generate-schema").json(
{array:[
    {"a2":[{something:""}]},
    {"a2":[]},
]})
```

The object inside a2 array should have the "something" field. But the result is now

```
{"a2":{"type":"array","items":{}}}
```

vs 

```
require("generate-schema").json(
{array:[
    {"a2":[{something:""}]}
]})
```
which correctly infers

```
{"a2":{"type":"array","items":{"type":"object","properties":{"something":{"type":"string"}}}}}
```

The current code handles non nested array nicely though.

```
require("generate-schema").json(
{array:[
    {"a2":[{something:""}]},
    {}

]})
```

generates
```
{"a2":{"type":"array","items":{"type":"object","properties":{"something":{"type":"string"}}}
```

**Comment by nijikokun** [comment]
Great catch, added fix for this, and a fix for different types in the nested array. This also handles mixed objects & types properly as well now

**Comment by nijikokun** [comment]
Released in [v2.4.0](https://github.com/nijikokun/generate-schema/releases/tag/v2.4.0)
